orkaudio/audiocaptureplugins/voip/VoIpSession.cpp

```diff
@@ -4503,6 +4504,113 @@ CStdString VoIpSessions::StartCaptureNativeCallId(CStdString& nativecallid, CStd
 	return orkUid;
 }
 
+CStdString VoIpSessions::StartStreamNativeCallId(CStdString& nativecallid)
+{
+	std::map<unsigned long long, VoIpSessionRef>::iterator pair;
+	bool found = false;
+	CStdString logMsg;
+	VoIpSessionRef session;
+	CStdString orkUid = CStdString("");
+
+	for(pair = m_byIpAndPort.begin(); pair != m_byIpAndPort.end() && found == false; pair++)
+	{
+		session = pair->second;
+
+		if(session->NativeCallIdMatches(nativecallid))
+		{
+			session->m_keepRtp = true;
+			found = true;
+			orkUid = session->GetOrkUid();
+		}
+	}
+
+	if(found)
+	{
+		CaptureEventRef event(new CaptureEvent());
+						event->m_type = CaptureEvent::EtKeyValue;
+						event->m_key = "LiveStream";
+						event->m_value = "start";
+						g_captureEventCallBack(event, session->m_capturePort);
+
+		logMsg.Format("[%s] StartStreamNativeCallId: Started capture, nativecallid:%s", session->m_trackingId, nativecallid);
+	}
+	else
+	{
+		logMsg.Format("StartStreamNativeCallId: No session has native callid:%s", nativecallid);
+	}
+
+	LOG4CXX_INFO(m_log, logMsg);
+
+	return orkUid;
+}
+
+CStdString VoIpSessions::EndStreamNativeCallId(CStdString& nativecallid)
+{
+	std::map<unsigned long long, VoIpSessionRef>::iterator pair;
+	bool found = false;
+	CStdString logMsg;
+	VoIpSessionRef session;
+	CStdString orkUid = CStdString("");
+
+	for(pair = m_byIpAndPort.begin(); pair != m_byIpAndPort.end() && found == false; pair++)
+	{
+		session = pair->second;
+
+		if(session->NativeCallIdMatches(nativecallid))
+		{
+			session->m_keepRtp = true;
+			found = true;
+			orkUid = session->GetOrkUid();
+		}
+	}
```

Comment:
Duplicate code in Start/End function, please refactor

Reply:
done